clients/src/main/java/org/apache/kafka/common/utils/KafkaLoggerFactory.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.utils;
+
+public class KafkaLoggerFactory {
```

Comment:
Btw, another alternative would be to introduce a `LogContext` containing a `Map<String, String>`. A bit like slf4j's `Marker`, but flat instead of hierarchical. The nice thing is that one could enrich the context, if necessary. One could eventually reuse it for metrics as well, but we don't have to figure that one out now. Thoughts?